shared/middleware/helpers/redis_property_bag.rb

```diff
@@ -0,0 +1,73 @@
+# A property bag implementation backed by Redis.
+
+# Implementation notes:  The entire property bag is stored in a single
+# value as a Redis hash. This makes it efficient to fetch tne entire property
+# bag without having to make multiple requests to Redis.
+
+require 'sinatra/base'
+require 'redis'
+
+class RedisPropertyBag
+
+  # A unique prefix for Redis property bag keys.
+  PROPERTY_BAG_KEY_PREFIX = "_pbag__"
+
+  class NotFound < Sinatra::NotFound
+  end
```

Comment:
I'm _guessing_ this is a stub for instrumenting metrics later or something, not just an alias?  If you are just aliasing, it may be cleaner to just assign the Sinatra class to a new constant.
```ruby
NotFound = Sinatra::NotFound
```
You only use `NotFound` once right now.  Do you expect to use it in other places, long-term?